bokehjs/src/coffee/models/plots/plot_canvas.coffee

```diff
@@ -246,46 +246,105 @@ class PlotCanvasView extends Renderer.View
   reset_selection: () ->
     @update_selection(null)
 
-  _update_single_range: (rng, range_info, is_panning) ->
-    # Is this a reversed range?
-    reversed = if rng.get('start') > rng.get('end') then true else false
-
-    # Prevent range from going outside limits
-    # Also ensure that range keeps the same delta when panning
-
-    if rng.get('bounds')?
-      min = rng.get('bounds')[0]
-      max = rng.get('bounds')[1]
-
-      if reversed
-        if min?
-          if min >= range_info['end']
-            range_info['end'] = min
-            if is_panning?
-              range_info['start'] = rng.get('start')
-        if max?
-          if max <= range_info['start']
-            range_info['start'] = max
-            if is_panning?
-              range_info['end'] = rng.get('end')
-      else
-        if min?
-          if min >= range_info['start']
-            range_info['start'] = min
-            if is_panning?
-              range_info['end'] = rng.get('end')
-        if max?
-          if max <= range_info['end']
-            range_info['end'] = max
-            if is_panning?
-              range_info['start'] = rng.get('start')
-
-    if rng.get('start') != range_info['start'] or rng.get('end') != range_info['end']
-      rng.have_updated_interactively = true
-      rng.set(range_info)
-      rng.get('callback')?.execute(rng)
+  _update_ranges_together: (range_info_iter) ->
+    # Get weight needed to scale the diff of the range to honor interval limits
+    weight = 1.0
+    for [rng, range_info] in range_info_iter
+      weight = Math.min(weight, @_get_weight_to_constrain_interval(rng, range_info))
+    # Apply shared weight to all ranges
+    if weight < 1
+      for [rng, range_info] in range_info_iter
+        range_info['start'] = weight * range_info['start'] + (1-weight) * rng.get('start')
+        range_info['end'] = weight * range_info['end'] + (1-weight) * rng.get('end')
+
+  _update_ranges_individually: (range_info_iter, is_panning, is_scrolling) ->
+    hit_bound = false
+    for [rng, range_info] in range_info_iter
+      # Is this a reversed range?
+      reversed = if rng.get('start') > rng.get('end') then true else false
```

Comment:
I know this was copied from the original but can't this just be: `reversed = (rng.get('start') > rng.get('end'))` ?